src/pipeline.cc

```diff
@@ -809,6 +809,9 @@ class PipelineWorker : public Nan::AsyncWorker {
             ->set("squash", baton->tiffSquash)
             ->set("compression", baton->tiffCompression)
             ->set("predictor", baton->tiffPredictor)));
+            ->set("predictor", baton->tiffPredictor)
```

Comment:
I also added xres, and yres when writing to buffer. However, these values do not get saved when calling the ToBuffer method on a tiff image.
e.g. when I write a test like this:
```
 it('TIFF setting xres and yres on buffer', function (done) {
    const res = 1000.0; // inputTiff has a dpi of 300 (res*2.54)
    const image = sharp(fixtures.inputTiff)
      .tiff({
        xres: (res),
        yres: (res)
      })
      .toBuffer(function (err, data, info) {
        if (err) throw err;
        sharp(data).metadata(function (err, metadata) {
          if (err) throw err;
          assert.strictEqual(metadata.density, res*2.54); // convert to dpi
          done();
        });
      });
  });
```
And then write it to disk afterwards or read the metadata, the density is still the same as the original(300dpi).
@lovell do you have any idea why?